app/src/main/java/com/github/polybooks/database/FBInterestDatabase.kt

```diff
@@ -179,20 +212,116 @@ class FBInterestDatabase: InterestDatabase {
-        TODO("Not yet implemented")
+    fun getLoggedUserInterests(user: LoggedUser): CompletableFuture<List<Interest>> {
+        val future = CompletableFuture<List<Interest>>()
+
+        FirebaseProvider.getFirestore()
+            .collection(userCollection)
+            .document(user.uid)
+            .get()
+            .addOnSuccessListener { document ->
+                if (document != null) {
+                   val mapList: List<Map<String, String>> = document.get(interestsTopicName) as List<Map<String, String>>
+                    val interestsList = mapList.map { interest ->
+                        when {
+                            interest.containsKey("name") -> {
+                                Topic(interest["name"] as String)
+                            }
+                            interest.containsKey("courseName") -> {
+                                Course(interest["courseName"] as String)
+                            }
+                            else -> {
+                                Semester(
+                                    section = interest["section"] as String,
+                                    semester = interest["semester"] as String
+                                )
+                            }
+                        }
+
+                    }
+                    future.complete(interestsList as List<Interest>)
+                } else {
+                    future.complete(emptyList())
+                }
+
+            }
+            .addOnFailureListener { future.completeExceptionally(DatabaseException("Could not retrieve the list of all interests of user ${user.uid} because of: $it")) }
+
+        return future
     }
 
     /**
-     * Sets the interests of the specified user.
+     * Sets the interests of the specified logged user.
      * TODO: Might need to add an authentication token to restrict authenticated users to only modify their interests.
      * @return A Future to receive confirmation of success/failure asynchronously
      * */
-    override fun setUserInterests(user: User, interests: List<Interest>): CompletableFuture<Unit> {
-        TODO("Not yet implemented")
+    fun setLoggedUserInterests(user: LoggedUser, interests: List<Interest>): CompletableFuture<List<Interest>> {
+        val future = CompletableFuture<List<Interest>>()
+
+        val docData = hashMapOf(
+            interestsTopicName to interests
+        )
+        FirebaseProvider.getFirestore()
+            .collection(userCollection)
+            .document(user.uid)
+            .set(docData, SetOptions.merge())
+            .addOnSuccessListener { future.complete(interests) }
+            .addOnFailureListener { future.completeExceptionally(DatabaseException("Failed to insert $docData of user ${user.uid} into Database because of: $it")) }
+
+        return future
+    }
+
+    // TODO list
+    // 2) use sharedPreferences for LoggedUser to serve as a cache.
```

Comment:
The TAs don't like when we put TODO lists in the code.

Reply:
I know, I thought about their comment when I wrote it ^^ But this one will disappear today in the next PR